oversweet-api/src/main/java/com/depromeet/oversweet/drink/vo/DrinkDailySugarInfo.java

```diff
@@ -0,0 +1,25 @@
+package com.depromeet.oversweet.drink.vo;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.swagger.v3.oas.annotations.media.Schema;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class DrinkDailySugarInfo{
+    @Schema(description = "해당 일")
+    @JsonProperty(value = "day")
+    private LocalDate day;
+    @Schema(description = "해당 요일에 먹은 당 섭취량")
+    @JsonProperty(value = "sugar")
+    private int sugar;
+
+    private DrinkDailySugarInfo(final LocalDateTime day, final int sugar) {
+        this.day = day.toLocalDate();
+        this.sugar = sugar;
+    }
+    public static DrinkDailySugarInfo of(final LocalDateTime day, final int sugar){
+        return new DrinkDailySugarInfo(day, sugar);
+    }
```

Comment:
DrinkStatisticsTotalInfo 의 getDayOfWeekSugarInfo() 에서 Map.entrySet을 통해서 key 값에 LocalDate 타입을 넣어주면 현재 LocalDateTime으로 받아서 또 다시 toLocalDate()을 호출할 이유가 없어 보여요

개인적으로 주생성자와 부생성자 개념을 적용하신다면
주생성자는 그 Class 객체 필드 그 자체로 초기화 하는 목적으로 사용하면 좋고

부생성자는 다양한 타입(LocalDateTime) 을 필드 타입 (LocalDate) 으로 변환하는 작업을 수행하고 주생성자를 호출하는 방향이 좀 더 이상적인 느낌이라고 생각해요
```suggestion
    // 주생성자
    private DrinkDailySugarInfo(final LocalDate day, final int sugar) {
        this.day = day;
        this.sugar = sugar;
    }
    
    // 부 생성자 (팩토리 메서드 활용)
    public static DrinkDailySugarInfo of(final LocalDate day, final int sugar){
        return new DrinkDailySugarInfo(day, sugar);
    }
```

추가적으로 개인적인 의견을 공유하자면
정적 팩토리 메서드는 DrinkDailySugarInfo 객체가 가지는 필드 그 이상으로 데이터를 받아서
DrinkDailySugarInfo Class을 초기화 할 때 바람직하다고 생각하고 있어요

예를 들어, 어떠한 이유 때문에 DrinkDailySugarInfo 에게 List<RecordEntity> 을 넘겨줘서 
DrinkDailySugarInfo 객체를 생성한다면
그때, DrinkDailySugarInfo 객체가 가지는 필드보다 그 이상으로 값을 받아서 객체 초기화를 해야하기에 정적 팩토리 메서드를 활용하면 가치있다 생각해요~

Reply:
이렇게 된다면 제 생각에는 레코드 클래스로 변경해도 될 거 같아요.

Reply:
이러한 경우에는 Record Class가 더 적절하다고 생각이 드네요